tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/CollectInformation.scala

```diff
@@ -330,4 +223,53 @@ object CollectInformation extends Logging {
     }
     allRows.filter(_.isDefined).map(_.get)
   }
-}
\ No newline at end of file
+
+  def printSQLPlans(apps: Seq[ApplicationInfo], outputDir: String): Unit = {
+    for (app <- apps) {
+      val planFileWriter = new ToolTextFileWriter(s"$outputDir/${app.appId}",
+        "planDescriptions.log", "SQL Plan")
+      try {
+        for ((sqlID, planDesc) <- app.physicalPlanDescription.toSeq.sortBy(_._1)) {
+          planFileWriter.write("\n=============================\n")
+          planFileWriter.write(s"Plan for SQL ID : $sqlID")
+          planFileWriter.write("\n=============================\n")
+          planFileWriter.write(planDesc)
+        }
+      } finally {
+        planFileWriter.close()
+      }
+    }
+  }
+
+  // updated processed properties hashmap based on the new rapids related
+  // properties passed in
+  def addNewProps(newRapidsRelated: Map[String, String],
+      processedProps: HashMap[String, ArrayBuffer[String]],
+      numApps: Int): Unit = {
+    val inter = processedProps.keys.toSeq.intersect(newRapidsRelated.keys.toSeq)
+    val existDiff = processedProps.keys.toSeq.diff(inter)
+    val newDiff = newRapidsRelated.keys.toSeq.diff(inter)
+
+    // first update intersecting
+    inter.foreach { k =>
+      val appVals = processedProps.getOrElse(k, ArrayBuffer[String]())
+      appVals += newRapidsRelated.getOrElse(k, "null")
```

Comment:
Sorry, I don't understanding this.  Could you please explain what does `appVals` here and below in existDiff used for? 
I see appVals being created locally but that is not being updated in any HashMap.

Reply:
so this is updating the values in processedProps itself.  appVals gets the Arraybuffer of properties that match the key.  ie key that has multiple values.  We adde any new values to that for that certain key.

This is because the table is backwards from other tables. Here the rows keys are the property keys and the app values are the columns.  So the ArrayBuffer here is the app values for each key.

We have to handle multiple cases to make sure all the array buffers for each key get updated with the same number of entries.  
1) key already existed
2) this app doesn't contain a key that other apps had
3) new key that wasn't in the other apps we processed

Reply:
Thanks for the explanation.  It makes sense. Didn't realize we were updating processedProps itself.